projects/angular/src/forms/input/input.spec.ts

```diff
@@ -39,5 +42,48 @@ export default function (): void {
     ControlStandaloneSpec(StandaloneUseTest);
     TemplateDrivenSpec(ClrInputContainer, ClrInput, TemplateDrivenTest, 'clr-input');
     ReactiveSpec(ClrInputContainer, ClrInput, ReactiveTest, 'clr-input');
+    inputSpec('tempate-driven', ClrInputContainer, ClrInput, TemplateDrivenTest);
+    inputSpec('reactive', ClrInputContainer, ClrInput, ReactiveTest);
+  });
+}
+
+function inputSpec(description, testContainer, testControl, testComponent) {
+  describe('Input specific value change tests ' + description, () => {
+    let control, fixture;
+
+    beforeEach(() => {
+      TestBed.configureTestingModule({
+        imports: [FormsModule, ClrCommonFormsModule, ReactiveFormsModule],
+        declarations: [testContainer, testControl, testComponent],
+      });
+      fixture = TestBed.createComponent(testComponent);
+      control = fixture.debugElement.query(By.directive(testControl));
+      fixture.detectChanges();
+    });
+
+    it('should handle valueChanges calls', () => {
+      // control must be both invalid and blurred to register the validity
+      let valueChanges = 0;
+      control.injector.get(NgControl).control.valueChanges.subscribe(() => {
+        valueChanges++;
+      });
+
+      // make sure blur alone does not trigger valueChanges
+      control.nativeElement.dispatchEvent(new Event('focus'));
+      control.nativeElement.dispatchEvent(new Event('blur'));
+      fixture.detectChanges();
+      expect(valueChanges).toBe(0);
+
+      // now make sure input change triggers valueChanges
+      control.nativeElement.dispatchEvent(new Event('focus'));
+      control.nativeElement.value = 'abc';
+      control.nativeElement.checked = true;
+      control.nativeElement.selected = true;
```

Comment:
What are these lines for? You can't check or select an `input` element.

Reply:
Absolutely right. This is relic from my unsuccessful attempts to make a common test in the `control.spec.ts`. 
Thanks, I'll remove them.